src/scenic/core/pruning.py

```diff
@@ -97,37 +131,76 @@ def pruneContainment(scenario, verbosity):
     bound the radius of O, then we can first erode C by that distance.
```

Comment:
Need to update docstring (now is more general since it allows uniform random + offset).